### PR TITLE
Homework of meeting 2

### PR DESCRIPTION
**Pull request by fredpantuzza**
Sorry I didn't do the first meeting version separately.

### CODE REVIEW

File: course7/homeworks/fred/ls/main.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      << pre << 'B';
+  return ss.str();
+}
+
+size_t getDirectorySize(const boost::filesystem::path& path)
+{
+  using namespace boost::filesystem;
+
+  assert(is_directory(path));
+  boost::filesystem::recursive_directory_iterator dir_iterator(path);
```
Comment (maciekgajewski) [inline course7/homeworks/fred/ls/main.cc:36]: Inconsistent qualification: you should be either using it or not

File: course7/homeworks/fred/ls/main.cc [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  return vm;
+}
+
+int main(int argc, char** argv)
+{
+  try {
+    auto arguments = parseArguments(argc, argv);
+  
+    //bool showDetails = arguments.count("long");
```
Comment (maciekgajewski) [inline course7/homeworks/fred/ls/main.cc:117]: what about this?

File: course7/homeworks/fred/ls/main.cc [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+    else {
+      std::cout << "More than one directory specified\n";
+      return 1;
+    }
+  
+    printDirectoryContents(directory, showTree);
+
+  } catch (const std::exception& ex) {
+    std::cout << ex.what() << std::endl;
```
Comment (maciekgajewski) [inline course7/homeworks/fred/ls/main.cc:135]: Don't do it; let the exception fall trough ,and produce nice, useful coredump

**Review by maciekgajewski — APPROVED** [review]
Cool. Using recursive iterator solves so many issues

File: course7/homeworks/fred/ls/main.cc (hunk 1 above)
Comment (fredpantuzza) [reply course7/homeworks/fred/ls/main.cc:36]: I didn't understand what you mean. Using what?

**Pull request merged**